app/src/main/java/org/listenbrainz/android/util/datastore/ProtoDataStore.kt

```diff
@@ -0,0 +1,78 @@
+package org.listenbrainz.android.util.datastore
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.map
+import org.listenbrainz.android.model.ResponseError
+import org.listenbrainz.android.util.Resource
+import org.listenbrainz.android.util.datastore.Preference.Companion.ComplexPreference
+import org.listenbrainz.android.util.datastore.Preference.Companion.PrimitivePreference
+import java.io.IOException
+
+abstract class ProtoDataStore(private val dataStore: DataStore<Preferences>) {
```

Comment:
@07jasjeet  I see you have made a library to do this. Can we import these from there instead of adding chunks of files to our codebase?

Reply:
Yes. On it.

Reply:
@07jasjeet any update on this?

Reply:
Will update by weekend.